include/ginkgo/core/base/exception_helpers.hpp

```diff
@@ -211,6 +211,24 @@ inline dim<2> get_size(const dim<2> &size) { return size; }
     }
 
 
+/**
+ * Asserts that _op1 can be applied to _op2.
```

Comment:
The documentation looks same as the macro above. Can you specify that this is for the outer dimension or something like that ?